pkg/scanner/local/scan.go

```diff
@@ -100,6 +101,76 @@ func (s Scanner) Scan(ctx context.Context, targetName, artifactKey string, blobK
 	return s.ScanTarget(ctx, target, options)
 }
 
+func postProcessApplications(apps []ftypes.Application, options types.ScanOptions) []ftypes.Application {
+	// Map to store nodejs lock file packages
+	nodeLockFilePackages := map[string]ftypes.Package{}
+	reqPHPPackages := make(map[string]struct{})
+	reqDevPHPPackages := make(map[string]struct{})
+
+	for i, app := range apps {
+		if len(app.Libraries) == 0 {
+			continue
+		}
+
+		isPkgSplitRequired := utils.IsPkgSplitRequired(app.Type)
+
+		// Get parents map for current target
+		parents := ftypes.Packages(app.Libraries).ParentDeps()
+
+		// Get app directory info for node
+		nodeAppDirInfo := utils.NodeAppDirInfo(app.FilePath)
+
+		for i, pkg := range app.Libraries {
+
+			// calculate rootDep
+			if len(parents) != 0 && (pkg.Indirect || isPkgSplitRequired) {
+				pkg.RootDependencies = utils.FindAncestor(pkg.ID, parents, map[string]struct{}{})
+				app.Libraries[i] = pkg
+
+				// if a pkg which direct and has atleast one root dependency
+				// we will consisder it as both direct and indirect
+				// so we append new entry for indirect and move root deps to the new entry
+				// note: for node-pkg this will done when we dedupe (at this point of time node-pkg won't have transitive info)
```

Comment:
Let's be explicit about the behaviour in case of image and fs scan. node-pkg is confusing.
Might be better if we as both cases image/fs scan and highlight behaviour